### ISSUE TITLE
Add `percentage-unless-within-keyword-only-block` primary option to keyframe-selector-notation

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#6191]
### What is the problem you're trying to solve?

I only want to disallow mixed keyframe selectors:

```css
@keyframes foo {
  from { opacity: 0; }
  50% { opacity: 0.5; }
  to { opacity: 1; }
}
```

```css
@keyframes foo {
  from { opacity: 0; }
  100% { opacity: 0.5; }
}
```

While allowing _either_ of the notations when not mixed:

```css
@keyframes foo {
  from { opacity: 0; }
  to { opacity: 1; }
}
```

```css
@keyframes foo {
  0% { opacity: 0; }
  100% { opacity: 1; }
}
```

Ref: https://github.com/stylelint/stylelint-config-standard/issues/255

### What solution would you like to see?

What's a good name for this option? Maybe:

```
ignore: ["after-same-notation"]`
```

We use `same` [else where](https://stylelint.io/user-guide/rules/list/at-rule-empty-line-before/#except-after-same-name-inside-block-blockless-after-same-name-blockless-blockless-after-blockless-first-nested) and the rule its scoped to the keyframe-selector rather than the keyframe-block.

**Comment by ybiquitous** [comment]
Indeed, a third primary option may be better, though I don’t have a good name for the option. 

For example, `percentage-unless-keyword-only`? But I'm not sure... 🤔

**Comment by jeddy3** [comment]
That's on the right track.

Is `percentage-unless-within-keyword-only-block` too verbose? The rule name is scoped to the keyframe selector, but we're basing the option on what's happening in the containing keyframe block. This name makes that explicit.

We use `within` [else where](https://stylelint.io/user-guide/rules/list/no-descending-specificity/#ignore-selectors-within-list) to show this relationship, and we've been verbose [else where](https://stylelint.io/user-guide/rules/list/declaration-block-no-duplicate-properties/#ignore-consecutive-duplicates-with-same-prefixless-values) too.

**Comment by ybiquitous** [comment]
`percentage-unless-within-keyword-only-block` is indeed long, but your rationale makes sense to me. 👍🏼
